download_weather_data.py

```
@@ -0,0 +1,286 @@
+        output_dir = output_dir or Path(f"data/{source}/processed")
+        output_dir.mkdir(parents=True, exist_ok=True)
+        
+        timestamp = datetime.now(timezone.utc).strftime("%Y%m%d_%H%M%S")
+        output_file = output_dir / f"{country.lower()}_{source}_weather_{timestamp}.csv"
+        
+        checkpoint_file = Path(f"data/{source}/checkpoints") / f"checkpoint_{country}_{source}_weather.json"
+        checkpoint_file.parent.mkdir(parents=True, exist_ok=True)
+        
+        checkpoint_data = {}
+        if checkpoint_file.exists():
+            with open(checkpoint_file, 'r') as f:
```

Ruff (0.12.2)

Async functions should not open files with blocking methods like `open`

(ASYNC230)